sectionedrecyclerview/src/main/java/com/cruxlab/sectionedrecyclerview/lib/SectionDataManager.java

```diff
@@ -256,9 +256,23 @@ private void removeSection(int section, boolean notify) {
     @Override
     public void updateSection(int section) {
         checkSectionIndex(section);
-        adapter.notifyItemRangeChanged(getSectionFirstPos(section), getSectionRealItemCount(section));
+        short sectionType = sectionToType.get(section);
+        SectionAdapterWrapper sectionAdapter = typeToAdapter.get(sectionType);
+        int oldItemsCount = getSectionRealItemCount(section);
+        int newItemsCount = sectionAdapter.getItemCount() + sectionAdapter.getHeaderVisibilityInt();
+        updatePosSum(section, newItemsCount, false);
```

Comment:
This method takes as a second parameter the value to be updated items count by, so you have to pass here `newItemsCount  - oldItemsCount`:
```suggestion
        updatePosSum(section, newItemsCount  - oldItemsCount, false);
```